crates/provider-sdk/src/provider.rs

```diff
@@ -196,133 +224,166 @@ impl ProviderConnection {
+
+                                        // Send the error forwards on the error subject
+                                        if let Err(err) = transmitter
+                                            .transmit_static(error_subject, format!("{err:#}"))
+                                            .await
+                                        {
+                                            error!(?err, "failed to transmit error to invoker");
+                                        }
+                                    },
+                                    Ok(bytes) => {
+                                        // Assuming that the provider has processed the request and produced objects
+                                        // that conform to wrpc, transmit the response that were returned by the invocation
+                                        if let Err(err) = transmitter.transmit(result_subject, bytes.into()).await {
+                                            error!(%err, "failed to transmit invocation results");
+                                        }
+                                    },
+                                };
+                            }
+                            Ok(())
+                        }
                 }
-            } /* loop */
-        });
-        Ok(handle)
+            }));
+        }
+
+        Ok(handles)
     }
 
-    async fn handle_rpc<P>(&self, provider: P, inv: Invocation) -> InvocationResult<Vec<u8>>
+    /// Handle an invocation coming from wRPC
+    ///
+    /// # Arguments
+    ///
+    /// * `provider` - The Provider
+    /// * `operation` - The operation being performed (of the form `<ns>:<pkg>/<interface>.<function>`)
+    /// * `source_id` - The ID of the origin which might represent one or more components/providers (ex. an actor public key)
+    /// * `wrpc_invocation` - Details of the wRPC invocation
+    async fn handle_wrpc<P>(
+        &self,
+        provider: P,
+        operation: impl AsRef<str>,
+        source_id: impl AsRef<str>,
+        invocation_params: Vec<wrpc_transport::Value>,
+    ) -> InvocationResult<Vec<u8>>
     where
         P: Provider + Clone,
     {
-        let inv = self.rpc_client.dechunk(inv).await?;
-        let (inv, claims) = self
-            .rpc_client
-            .validate_invocation(inv)
-            .await
-            .map_err(InvocationError::from)?;
-        self.validate_provider_invocation(&inv, &claims)
-            .await
-            .map_err(InvocationError::from)?;
-        let span = tracing::debug_span!("dispatch", public_key = %inv.origin.public_key, method = %inv.operation);
+        let operation = operation.as_ref();
+        let source_id = source_id.as_ref();
+
+        // todo(vadossi-cosmonic): do we need to do the stuff that used to happen here? (see handle_rpc())
+        // - dechunking (we still need to do this if the wrpc_transport::Value is big enough, right??) NATS is still gonna take chunks
```

Comment:
wrpc-transport handles chunking, if a message of size S is to be sent on topic A and it exceeds the byte size limit N, then S/N messages will be sent on topic A.
E.g. with a hypothetical limit of 1 byte and a payload of `abc`, exactly 3 messages will be sent - `a`, `b` and `c`

Clients are free to arbitrarily chunk data as they see fit